pennylane/pulse/parametrized_evolution.py

```diff
@@ -142,6 +156,11 @@ def circuit(params):
         operators are executed simultaneously, but rather that both evaluate their respective
         scalar-valued functions using the same time window. See Usage Details.
 
+    .. note::
+
+        Using ``return_intermediate`` in a quantum circuit leads to broadcasted execution,
+        which can lead to unintended additional computational cost.
+        Also consider the usage details below.
```

Comment:
Could it make sense to provide the `return_intermediate` and `complementary` flags in the `matrix()` method instead of the `ParametrizedEvolution` itself?

Reply:
I was thinking the same, but somehow we need to be able to provide this at creation for the shift rule :grimacing: